pep-0710.rst

```diff
@@ -0,0 +1,461 @@
+give more accurate reports. Yet another use case could be reconstruction of the
+Python environment by pinning each installed package to a specific distribution
+artifact consumed from a Python package index.
+
+Rationale
+=========
+
+The motivation described in this PEP is an extension of that in :pep:`610`.
+In addition to recording provenance information for packages installed using a direct URL,
+installers should also do so for packages installed by name
+(and optionally version) from Python package indexes.
+
+The idea described in this PEP originated in a tool called `micropipenv
+<https://github.com/thoth-station/micropipenv>`__ that is used to install
+:term:`distribution packages <Distribution Package>` in containerized
+environments (see the reported issue `thoth-station/micropipenv#206`_).
+Currently, the assembled containerized application does not implicitly carry
+information about the provenance of installed distribution packages
+(unless these are installed from full URLs and recorded via ``direct_url.json``).
+This requires container image suppliers to link
+container images with the corresponding build process, its configuration and
+the application source code for checking requirements files in cases when
+software present in containerized environments needs to be audited.
+
+The `subsequent discussion in the Discourse thread
+<https://discuss.python.org/t/12316>`__ also brought up
+pip's new ``--report`` option that can
+`generate a detailed JSON report <pip_installation_report_>`__ about
+the installation process. This option could help with the provenance problem
+this PEP approaches. Nevertheless, this option needs to be *explicitly* passed
+to pip to obtain the provenance information, and includes additional metadata that
+might not be necessary for checking the provenance (such as Python version
+requirements of each distribution package). Also, this option is
+specific to pip as of the writing of this PEP.
+
+Note the current :ref:`spec for recording installed packages
+<packaging:recording-installed-packages>` defines a ``RECORD`` file that
+records installed files, but not the distribution artifact from which these
+files were obtained. Auditing installed artifacts can be performed
+based on matching the entries listed in the ``RECORD`` file. However, this
+technique requires a pre-computed database of files each artifact provides or a
+comparison with the actual artifact content. Both approaches are relatively
+expensive and time consuming operations which could be eliminated with the
+proposed ``provenance_url.json`` file.
+
+Recording provenance information for installed distribution packages,
+both those obtained from direct URLs and by name/version from an index,
+can simplify auditing Python environments in general, beyond just
+the specific use case for containerized applications mentioned earlier.
+environments. A community project `pip-audit
```

Comment:
Oops, suggestion-editing mistake on my part, sorry

(FWIW, this one of the reasons why I find SemBr, or at the very least avoiding spliced sentences, to be much easier to edit, review and diff than fixed hard breaks).

Reply:
No worries at all!

I'll use SemBr in next PEPs if I'll have a chance to write one again - kept the current style if that's fine. Sorry for any additional work it requires on your end.

Reply:
> kept the current style if that's fine.

If you want me to change it, please let me know. I kept it because the rest of the PEP uses it, nevertheless I can find some time for it if it would make things better.